version: '3.4'
services:
  front:
    container_name: app
    build:
      context: ./
      dockerfile: Dockerfile
    restart: on-failure
    ports:
      - 80:80
      - 443:443
    volumes:
      - ./app:/srv/app
      - ./certbot/www:/var/www/certbot/:ro
      - ./certbot/conf/:/etc/nginx/ssl/:ro
    tty: true
    depends_on:
      - back
  certbot:
    image: certbot/certbot:latest
    volumes:
      - ./certbot/www/:/var/www/certbot/:rw
      - ./certbot/conf/:/etc/letsencrypt/:rw
  back:
    container_name: backend
    build:
      args:
        - NODE_ENV=prod
        - DBUSERNAME=${DBUSERNAME}
        - PASSWORD=${PASSWORD}
        - DATABASE=${DATABASE}
        - HOSTNAME=${HOSTNAME}
        - DATABASE_PORT=${DATABASE_PORT}
        - SECRET=${SECRET}
      context: ../back
      dockerfile: Dockerfile
    restart: on-failure
    ports:
      - '3001:3001'
      - '8080:8080'
    environment:
      - NODE_ENV=prod
      - DBUSERNAME=${DBUSERNAME}
      - PASSWORD=${PASSWORD}
      - DATABASE=${DATABASE}
      - HOSTNAME=${HOSTNAME}
      - DATABASE_PORT=${DATABASE_PORT}
      - SECRET=${SECRET}
    volumes:
      - uploads:/public/uploads
      - ../front/certbot/www:/var/www/certbot/:ro
      - ../front/certbot/conf/:/etc/nginx/ssl/:ro
    depends_on:
      - database
  database:
    image: postgres:15.3-alpine
    volumes:
      - database:/var/lib/postgresql/data
    restart: on-failure
    ports:
      - 5432:5432
    environment:
      - POSTGRES_PASSWORD=${PASSWORD}
volumes:
  database:
  uploads: